tests/cpp/test_gpu1.cpp

```diff
@@ -5388,180 +5388,169 @@ TEST_F(NVFuserTest, FusionReductionSchedulerMultiDimFastest_CUDA) {
+    });
 
-      TensorView* tv1_cast = tv1;
-      if (is_fp16) {
-        tv1_cast = castOp(DataType::Half, tv1);
-      }
-      if (is_bf16) {
-        tv1_cast = castOp(DataType::BFloat16, tv1);
-      }
+// data type, reduction dim size, iteration dim size, reduction axis
+using ReductionWithIterDimParams =
+    std::tuple<DataType, int64_t, int64_t, int64_t>;
+using ReductionWithIterDimShmoo =
+    NVFuserFixtureParamTest<ReductionWithIterDimParams>;
+TEST_P(ReductionWithIterDimShmoo, FusionReductionSchedulerDimShmoo_CUDA) {
+  auto [dtype, rdim, odim, axis] = GetParam();
 
-      fusion.addOutput(tv1_cast);
+  at::ScalarType aten_dtype = data_type_to_aten(dtype);
 
-      auto options = at::TensorOptions().dtype(aten_dtype).device(at::kCUDA, 0);
+  // Shmoo tests can occupy a lot of memory due to allocating many
+  // different tensor sizes. So in order to avoid an OOM during this
+  // test, we manually clear the allocator after it's reached a certain
+  // threshold.
+  maybeClearAllocator();
 
-      at::Tensor aten_input = at::randn({rdim}, options);
-      auto aten_output = aten_input.to(at::kDouble).sum({0});
+  Fusion fusion;
+  FusionGuard fg(&fusion);
 
-      auto reduction_params = getReductionHeuristics(&fusion, {aten_input});
-      NVF_CHECK(reduction_params != nullptr, "Reduction is not found!");
-      scheduleReduction(&fusion, *reduction_params);
-      auto lparams = reduction_params->lparams;
+  bool is_fp16 = dtype == DataType::Half;
+  bool is_bf16 = dtype == DataType::BFloat16;
 
-      FusionExecutor fe;
-      fe.compileFusion(&fusion, {aten_input}, lparams);
-      auto cg_outputs = fe.runFusion({aten_input}, lparams);
-
-      testValidate(
-          &fusion,
-          cg_outputs,
-          {aten_input},
-          {aten_output},
-          __LINE__,
-          __FILE__,
-          "",
-          lparams);
-    }
-  }
-}
+  TensorView* tv0 = makeSymbolicTensor(2, dtype);
+  fusion.addInput(tv0);
 
-TEST_F(NVFuserTest, FusionReductionSchedulerDimShmoo_CUDA) {
-  std::vector<DataType> dtypes = {
-      DataType::Double,
-      DataType::Float,
-      DataType::Half,
-      DataType::ComplexFloat,
-      DataType::ComplexDouble};
-#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
-  if (at::cuda::getDeviceProperties(0)->major >= 8) {
-    dtypes.insert(dtypes.end(), DataType::BFloat16);
+  TensorView* tv0_cast = tv0;
+  if (is_fp16 || is_bf16) {
+    tv0_cast = castOp(DataType::Float, tv0);
   }
-#endif
 
-  std::vector<int64_t> red_axis = {1, 0};
-  std::vector<int64_t> output_dims = {160, 320};
-  std::vector<int64_t> red_dims;
+  TensorView* tv1 = sum(tv0_cast, {axis});
 
-  // Tried to cut down the number iterations with just
-  // doing every other power of 2.
-  for (int64_t i = 1; i <= 1024 * 1024; i <<= 2) {
-    red_dims.push_back(i);
+  TensorView* tv1_cast = tv1;
+  if (is_fp16) {
+    tv1_cast = castOp(DataType::Half, tv1);
+  }
+  if (is_bf16) {
+    tv1_cast = castOp(DataType::BFloat16, tv1);
```

Comment:
Same as above.

Reply:
On second thought, I now see that it was like this before this change. Better to not make changes like that in this PR then.